### PR TITLE
feat: support running tests in multiple OS versions

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It adds OSX and Windows to the build matrix in Travis when running the tests. For GH actions, it adds windows workers (unfortunately macos workers don't support Docker).

To avoid writing scripts in both BATCH and BASH, I'm adding support for [Mage](https://magefile.org/) as build system.

## Why is it important?
We want to build the library in different OS to potentially detect bugs or unexpected behaviours

## Related issues
- Relates to #232
- Relates #152

## Screenshots
The Travis CI job will show as this:
Screenshot 2020-08-24 at 08 56 28

The GH action job will show as this:
Screenshot 2020-11-11 at 13 18 20

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//+build mage
+
+package main
+
+import "github.com/magefile/mage/sh"
+
+// Runs go fmt
+func Format() error {
+	return sh.Run("go", "fmt", "./...")
```
Comment (mdelapenya) [inline magefile.go:9]: At this moment, this is not returning an error even when formatting files, so Travis won't be notified. I'm currently investigating why

**Comment by mdelapenya** [comment]
I forgot why Travis (and many CIs) does not support Docker for Mac in an easy manner, as it's needed to open Docker for Mac manually 🤦 I'll probably close this PR, unless anybody here find a workaround

**Comment by gianarb** [comment]
Yeah, I do not really have anything to suggest to about it. I think testcontainers-java uses GitHub Actions now. Maybe we can look at that as well.

**Comment by mdelapenya** [comment]
@gianarb which stages should we run on CI and in which platforms?

File: go.mod [hunk 2]
```diff
@@ -38,3 +39,5 @@ require (
 	gopkg.in/yaml.v2 v2.3.0
 	gotest.tools v0.0.0-20181223230014-1083505acf35
 )
+
+replace golang.org/x/sys => golang.org/x/sys v0.0.0-20190830141801-acfa387b8d69
```
Comment (mdelapenya) [inline go.mod:43]: This is what others are doing 🤷  https://github.com/golang/go/issues/34610

**Pull request closed without merging**